### DISCUSSION TITLE
Can I use C++ to write an operating system?

### DISCUSSION THREAD

**Discussion by IshikaMahajan7** [community/community#42253] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

![download](https://user-images.githubusercontent.com/115925848/208819630-ac8bbf31-5e2a-4f01-8069-e2f8b9496c66.jpeg)

**Comment by chandrakant22** [comment]
Yes, of course.

I suppose I should add a qualification to that: although a number of operating systems (e.g., BeOS, Chorus, Amoeba, Haiku, Syllable) have been written primarily in C++, all operating systems of which I'm aware include at least a few small bits and pieces written in assembly language. This is true almost regardless of the higher-level language chosen for the core of the operating system though.
